doc/stp/MSTP_HLD.md

```diff
@@ -0,0 +1,1057 @@
+PortNum           BPDU Tx     BPDU Rx     TCN Tx     TCN Rx             
+Ethernet13        10	      4           3          4
+PortChannel15     20	      6           4          1
+```
+
+## Clear Commands
+
+- sonic-clear spanning_tree mst statistics 
+- sonic-clear spanning_tree mst statistics instance \<instance-id\>
+
+
+## Debug Commands
+Following debug commands will be supported for enabling additional logging which can be viewed in /var/log/stpd.log, orchagent related logs can be viewed in /var/log/syslog.
+
+- debug spanning_tree mst instance \<instance-id\>
+- debug spanning_tree mst bpdu [tx|rx]
+- debug spanning_tree mst event
+- debug spanning_tree mst verbose
+- debug spanning_tree mst interface \<ifname>\
+
+Following debug commands will be supported for displaying internal data structures
+
+- debug spanning_tree mst dump instance \<instance-id\>
+
+
+## Disabled Commands
+Following commands are used to configure parameters at VLAN level and these commands are disabled if spanning-tree mode is `mstp`:
+
+- config spanning_tree mst vlan \<vlan-id\> (enable|disable) 
+- config spanning_tree mst vlan \<vlan-id\> forward_delay  \<fwd-delay-value\>
+- config spanning_tree mst vlan \<vlan-id\> hello \<hello-value\>
+- config spanning_tree mst vlan \<vlan-id\> max_age \<max-age-value\>
+- config spanning_tree mst vlan \<vlan-id\>priority  \<priority-value\>
+- config spanning_tree mst vlan \<vlan-id\> interface \<ifname\> cost \<value\>
+- config spanning_tree mst vlan \<vlan-id\> interface \<ifname\> priority \<value\>
+
+
+# YANG Model
+
+YANG Model will be extended as follows for MSTP:
+```yang
+module sonic-stp {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/sonic-stp";
+    prefix "stp";
+
+    import sonic-port {
+        prefix port;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+        revision-date 2021-06-13;
+    }
+
+    import sonic-vlan {
+        prefix vlan;
+        revision-date 2021-04-22;
+    }
+
+    import sonic-device_metadata { 
+        prefix device_metadata;
+        revision-date 2021-02-27;
+    }
+
+    description "STP yang Module for SONiC OS";
+
+    revision 2023-04-18 {
+        description "First Revision";
+    }
+
+    grouping interfaceAttr {
+        leaf path_cost {
+            type uint64 {
+                range "1..200000000" {
+                    error-message "Invalid Port Path Cost value.";
+                }
+            }
+            default 200;
+            description
+                "The port's contribution, when it is the Root Port,
+                to the Root Path Cost for the Bridge";
+        }
+
+        leaf priority {
+            type uint8 {
+                range "0..240" {
+                    error-message "Invalid Port Priority value.";
+                }
+            }
+            default 128;
+            description
+                "The manageable component of the Port Identifier,
```

Comment:
Can you please add the information that priority values should be increments of 16 in the description?